File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int minprice = prices[0];
        int ans = 0;
        for(int i=0;i<prices.size();i++){
            if(prices[i] < minprice) minprice = prices[i];
            else{
                ans = max(ans,prices[i] - minprice);
            }
        }
        return ans;
    }
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        minprice = prices[0]
        ans = 0
        for i in range(len(prices)):
            if(prices[i] < minprice):
                minprice = prices[i]
            else:
                ans = max(ans,prices[i] - minprice)
        return ans